integ/util.go

```diff
@@ -67,6 +67,7 @@ type IwfServiceTestConfig struct {
 	MemoEncryption                   bool
 	DisableFailAtMemoIncompatibility bool // default to false so that we will fail at test
 	OptimizedVersioning              *bool
+	DefaultHeaders                   map[string]string
```

Comment:
I saw that the previous implementation had integration tests that would check for the presence of the headers in every call, am I seeing that right? https://github.com/indeedeng/iwf/pull/356/files#diff-c3752f9aebfa9a670e62a5c0cc34c36bbfbfcbe849288ece0973806a34eac156R30-R33

I wanted to do something more flexible, where we would have a specific integration test for checking the headers passed in the configuration are being used, rather than having the route check it every time a call was made. 

I do need some of your guidance here, I can't figure out how to do the integration tests with this approach.

Reply:
The headers will be used when making calls for stateAPI. So for a specific integ tests, you will still need to build out a router. But it will be cleaner to have that specific router, is that what you meant?

Reply:
That's what I'm thinking yes. Do you think that would be overkill for testing such a small feature?

Reply:
I think it’s a good idea. I have been lazy to do the right thing and the tests have become messy here

Reply:
I have created the tests now, I struggled a bit navigating through the existing ones trying to use one as a template. I'm sure these can be better. We can have a chat tomorrow about them and you can point me where I've gone wrong! Thanks in advance! 🙌

Reply:
Sounds good. I went through it and left you some comments already. and we can go through them later